src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -129,38 +128,44 @@
         this.logsProcessor = logProcessor;
         this.executorService = executorService;
 
-        topics.lookup(CONFIGURATION_CONFIG_KEY, LOGS_UPLOADER_PERIODIC_UPDATE_INTERVAL_SEC)
-                .dflt(DEFAULT_PERIODIC_UPDATE_INTERVAL_SEC)
-                .subscribe((why, newv) -> {
-                    if (why == WhatHappened.removed) {
-                        periodicUpdateIntervalSec = DEFAULT_PERIODIC_UPDATE_INTERVAL_SEC;
-                    } else {
-                        if (Coerce.toInt(newv) > 0) {
-                            periodicUpdateIntervalSec = Coerce.toInt(newv);
-                        } else {
-                            logger.atWarn().log("Invalid config value, {}, for periodicUploadIntervalSec. Must be an "
-                                    + "integer greater than 0. Using default value of 300 (5 minutes)",
-                                    Coerce.toInt(newv));
-                            periodicUpdateIntervalSec = DEFAULT_PERIODIC_UPDATE_INTERVAL_SEC;
-                        }
-                    }
-                });
+        topics.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((why, newv) -> {
+            logger.atWarn().log("Log manager config change. Why: {}, Node: {}", why, newv);
+            handlePeriodicUploadIntervalSecConfig(topics);
+            handleLogsUploaderConfig(topics);
```

Comment:
subscriptions can fire more often then you think -`why` can just be a time stamp updated and the value doesn't change. This can happen around deployments.

Will this behavior (no actual diff in value) cause any extra work to occur?

Reply:
Updated to exclude WhatHappened.timestampUpdated. The other options for WhatHappened seem to involve potential value changes.